File: Week_01/OnePreDay/MoveZeroes.js
var moveZeroes = function(nums) {
    let i = 0;
    let index = 0;
    for (index; index < nums.length; index++) {
        if (nums[index] != 0) {
            if (index != i) {
                nums[i] = nums[index];
                nums[index] = 0;
            }
            i++;
        }
    }
    return nums;
};
console.log(moveZeroes([0,1,0,3,12]));

File: Week_01/Homework/MoveZeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int i = 0;
        int len = nums.length;
        for (int j = 0; j < len; j++) {
            if (nums[j] != 0) {
                if (i != j) {
                    int temp = nums[i];
                    nums[i] = nums[j];
                    nums[j] = temp;
                }
                i++;
            }
        }
    }
}